examples/DotNetCore/BackupStore/BackupFromStorageQueue.cs

```diff
@@ -0,0 +1,146 @@
+
+                // Connect to storage queue
+                CloudStorageAccount storageAccount = CloudStorageAccount.Parse(Environment.GetEnvironmentVariable("StorageQueueConnectionString"));
+                CloudQueueClient queueClient = storageAccount.CreateCloudQueueClient();
+                CloudQueue queue = queueClient.GetQueueReference("eventgridqueue");
+                CloudQueueMessage customMessage = new CloudQueueMessage($"Refreshing Azure App Configuration store at: {DateTime.Now}.");
+
+                // Peek at the next message
+                CloudQueueMessage peekedMessage = queue.PeekMessage();
+                while (peekedMessage != null)
+                {
+                    var messageBody = peekedMessage.AsString;
+                    try
+                    {
+                        JObject dataObject = JObject.Parse(messageBody);
+                        string topic = dataObject.Property("topic")?.Value.ToString();
+                        if (topic != null && topic == Environment.GetEnvironmentVariable("PrimaryStoreResourceId"))
+                        {
+                            // We have events that need to be processed
+                            // Add custom message to the end of queue
+                            queue.AddMessage(customMessage);
+                            break;
+                        }
+                    }
+                    catch (JsonReaderException)
+                    {
+                        // If its not a valid JSON, ignore the message 
+                        // because it will be discarded from the queue
+                    }
+                    log.LogInformation($"Deleting message from queue: {messageBody}");
+                    queue.DeleteMessage(queue.GetMessage());
+                    peekedMessage = queue.PeekMessage();
+                }
+
+                // return if there are no messages in the queue
+                if (peekedMessage == null)
+                {
+                    log.LogInformation($"No events received for processing.");
+                    return;
+                }
+
+                // Read all settings from Primary Store and create local copy
+                IEnumerable<ConfigurationSetting> primaryStoreSettings = primaryAppConfigClient.GetConfigurationSettings(new SettingSelector());
+                IDictionary<Tuple<string, string>, ConfigurationSetting> primaryStoreLocalSettings = new Dictionary<Tuple<string, string>, ConfigurationSetting>();
+                foreach (ConfigurationSetting setting in primaryStoreSettings)
+                {
+                    var keyLabel = new Tuple<string, string>(setting.Key, setting.Label);
+                    primaryStoreLocalSettings.Add(keyLabel, setting);
+                }
+
+                // Read all settings from secondary store and create local copy
```

Comment:
Instead of fetching all configuration settings from both primary and secondary stores, how about fetching settings only from primary store and blindly updating the secondary store?

Reply:
After thinking a bit more on this, it seems this might be better in cases where we process a queue message but don't detect any changes in the settings. In this case, we would only read settings but not write them.

Reply:
Yes. Consider the case when you have 10 update events in the queue, and you add custom message at 11th position in queue. Before you have fetched all settings from primary store, 11th update event was added to the queue on 12th position.
This 11th event will not be deleted from the queue even though we will update it in the secondary store. So the next time Azure function runs, it may have to process just this one event which has actually already been updated in secondary store.